design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,455 @@
+
+To handle the missing **Job** granularity there are 2 options:
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Keep using only **namespace** and/or **service account** granularities. | \- Since Secrets Provider runs as a separate entity, we suggest creating a dedicated service account for Secrets Provider.<br />That way, the **service account** granularity serves as an exact match and **Job** granularity is redundant<br />- No code changes | \- Service account might be reused, which allows other apps to authenticate using Secrets Provider host<br />- Break application identity convention | Free              |
+| 2    | Add support for **Job** granularity in Conjur                | Allows specific identification of the Secrets Provider **Job**, which enhances security | \- Costly; requires code changes, verify compatibility, tests, docs<br />- Redundant if **service account** serves only the Secrets Provider | 10 days           |
+
+*Decision*: Solution #1, use **namespace** and **service account** to define the Secrets Provider host in Conjur policy. We would recommend to the customer to create a dedicated service account for Secrets Provider.
+
+#### Customer experience
+
+Deployment for the Secrets Provider will be done using Helm. That way, if the 
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined
+
+2. Create `values.yaml` that will contain the following parameters and their defaults:
+
+   For a full list of required parameters, see `values.yaml` below
+
+   *When parameters are not configured by customer, their defaults are used*
+
+3. Install Helm Chart for the Secrets Provider
+
+We will supply the customer with a  `values.yaml` file where all default parameters are collected. Our `values.yaml` will resemble the following:
+
+```
+# K8s permissioning defaults
+permission:
+	serviceAccountName: secrets-provider-account
+	roleName: secrets-provider-role
+	roleBinding: secrets-provider-role-binding
+
+# Secrets Provider pod defaults
+secrets-provider:
+	image: cyberark/secrets-provider-for-k8s
+	name: cyberark-secrets-provider
+	namespace: default
+	ttlSecondsAfterFinished: 0	# Length at which Job should stay alive even after finishing task. Used for logging.
+	
+# Secrets Provider environment variable defaults
+secrets-provider-environment:
+	conjurApplianceUrl: https://conjur-follower.default.svc.cluster.local
+	conjurAuthnUrl: https://conjur-follower.default.svc.cluster.local/authn-k8s/authn-default
+	conjurAuthnLogin: host/conjur/authn-k8s/authn-default/default
+	conjurAccount: default
```

Comment:
does it make sense to have these as default values? do we see a use-case in which the account will be `default`, the service-id will be `authn-default` or the host-id will be `default`?

Reply:
I do think that there should be defaults for these values. Customers take what is int he `values.yaml` file and copy them over to their customer values file. So if anything, it is a guide of what the customer needs to configure.

@orenbm  Is that your concern or do you agree that there should be defaults and just not agree with the values I gave to the defaults?

@oburstein-hub do you agree? do all configurations need to have defaults?

Reply:
I think that these fields do need defaults as they appear here - because it is the most typical case

Reply:
i don't think we should have defaults for these values. What's the common way to state required fields that don't have defualt values?

Reply:
There are values that are defaults and values that are mandatory. Those with defaults should be in `values.yaml` and those that are mandatory OR a customer wants to override the defaults should be in `custom-values.yaml`. I will add that distinction